lib/sprockets/less/v2/importer.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: true
+
+module Sprockets
+  module Less
+    module V2
+      # class used for importing files from SCCS and SASS files
+      class Importer
+        GLOB = /\*|\[.+\]/
+        IMPORT_SCANNER = /^\s*@import\s*['"]([^'"]+)['"]\s*;/.freeze
+
+        def dependencies
+          @dependencies ||= []
+        end
+
+        def context
+          @context
+        end
+
+        def fetch_import_paths(data)
+          data.scan(Regexp.union(IMPORT_SCANNER, GLOB)).flatten.compact.uniq
+        end
+
+        def fetch_all_dependencies(data, filename, less_options, css_options)
+          process_dependencies(data, filename, less_options, css_options)
+          [data, dependencies]
+        end
+
+        #Assemble dependencies for the context
+        def process_dependencies(data, filename, less_options, css_options)
+          @context = context = less_options[:custom][:sprockets_context]
+
+          final_css = ''.dup
+          import_paths = fetch_import_paths(data)
+          import_paths.each do |path|
+            css = find_relative(path, filename, less_options, css_options)
+            data.gsub!(/^\s*@import\s*['"]#{Regexp.escape(path)}['"]\s*;/, css)
+            final_css << css
+          end
+          final_css
```

Comment:
Use `each_with_object`, removing need for `final_css` and `import_paths` variables